### PR TITLE
Added SDK calls for federatedIdentityCred

### PR DESCRIPTION
**Pull request by Vidya2606**
# Description

Added SDK calls for federatedIdentityCred

Fixes # (issue)
Feature # (details)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration. Is it a  breaking change which will impact consuming tool(s).

- [ ] Test A
- [ ] Test B

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: pkg/cred/cred.go [hunk 1]
```diff
@@ -22,3 +24,20 @@ func GetCred() (*azidentity.DefaultAzureCredential, error) {
...
 	return cred, nil
 }
+
+func GetGraphCred() (*azidentity.DeviceCodeCredential, error) {
+	if graphCred != nil {
+		return graphCred, nil
+	}
+
+	gCreds, _ := azidentity.NewDeviceCodeCredential(&azidentity.DeviceCodeCredentialOptions{
+		TenantID: "", // todo - set this value
```
Comment (davidgamero) [inline pkg/cred/cred.go:34]: are these supposed to still be blank with todos?

Comment (Vidya2606) [reply pkg/cred/cred.go:34]: I think I'm supposed to pass the values, I had to figure out how to pass them.

File: pkg/providers/azure.go [hunk 2]
```diff
@@ -254,61 +255,91 @@ func (sc *SetUpCmd) ValidateSetUpConfig() error {
...
-		log.Errorf("error getting fic: %s", err)
-		return false
-	}
 
-	var fics []interface{}
-	if err = json.Unmarshal(out, &fics); err != nil {
-		log.Errorf("error marshaling fics: %s", err)
+	client := sc.AzClient.GraphClientFics
+
+	federatedIdentityCredentials, err := client.Applications().ByApplicationId(sc.appId).FederatedIdentityCredentials().ByFederatedIdentityCredentialId("").Get(ctx, nil)
```
Comment (davidgamero) [inline pkg/providers/azure.go:263]: does this filter for FederatedIdentityCredentials with an empty ID?

File: pkg/providers/az-client.go [hunk 3]
```diff
@@ -10,12 +10,14 @@ import (
...
 
 	msgraph "github.com/microsoftgraph/msgraph-sdk-go"
+	graphapp "github.com/microsoftgraph/msgraph-sdk-go/applications"
 )
 
 type AzClient struct {
 	AzTenantClient   azTenantClient
 	GraphClient      GraphClient
 	RoleAssignClient RoleAssignClient
+	GraphClientFics  GraphClientFics
```
Comment (davidgamero) [inline pkg/providers/az-client.go:20]: if these are both `GraphServiceClient` we can just use the GraphClient right?

**Review by davidgamero — CHANGES_REQUESTED** [review]

**Comment by Vidya2606** [comment]
The migration requires registration to Graph App which means the user had to do a device code login in the web browser which seemed to be a bad user experience, so we decided to keep the functions as is.

**Pull request closed without merging**